### PR TITLE
Tests: upcast `logits` to `float()`

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

Some of our `forward` numerical checks compare `logits` against expected values. #33902 removes the upcast scheduled for deprecation in `forward` (`logits = logits.float()`), resulting in significant memory savings.

However, our tests were expecting the cast. This PR is a fix for (some of) those tests. e.g. two slow tests in `llama` are fixed as a result of this PR.

(text search -> `EXPECTED_SLICE`, used in tests that compare `logits` against hardcoded values -> fix whenever applicable )

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
Sounds fair to me! Thanks @gante
